lib/phoenix/router.ex

```diff
@@ -775,10 +775,16 @@ defmodule Phoenix.Router do
     router_opts = Keyword.put(router_opts, :as, nil)
 
     quote unquote: true, bind_quoted: [path: path, plug: plug] do
-      plug = Scope.expand_alias(__MODULE__, plug)
-      path_segments = Route.forward_path_segments(path, plug, @phoenix_forwards)
-      @phoenix_forwards Map.put(@phoenix_forwards, plug, path_segments)
-      unquote(add_route(:forward, :*, path, plug, plug_opts, router_opts))
+      case plug do
+        mod when is_atom(mod) ->
```

Comment:
Theoretically we want to do as little work as possible inside the quoted expression. That's because this code will be compiled for every invocation and executed over and over again. So my suggestion for now is, can we put this code inside `Scope.expand_alias`?

Alternatively, we could move the whole mangling of `@phoenix_forwards` to a private function and raise there. It would also be the most performant approach too.